src/Summoner/CLI.hs

```diff
@@ -0,0 +1,181 @@
+
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+
+----------------------------------------------------------------------------
+-- CLI
+----------------------------------------------------------------------------
+
+summon :: IO ()
+summon = execParser prsr >>= runWithOptions
+
+-- | Run 'hs-init' with cli options
+runWithOptions :: InitOpts -> IO ()
+runWithOptions (InitOpts projectName targets) = do
+     -- Generate the project.
+    generateProject projectName targets
+
+    bold
+    T.putStrLn "\nJob's done"
+    reset
+
+
+
+-- | Initial parsed options from cli
+data InitOpts = InitOpts Text    -- ^ Project name
+                         Targets -- ^ Target flags
+
+targetsP ::  Decision -> Parser Targets
+targetsP d = do
+    githubFlag   <- githubP    d
+    travisFlag   <- travisP    d
+    appVeyorFlag <- appVeyorP  d
+    privateFlag  <- privateP   d
+    scriptFlag   <- scriptP    d
+    isLibrary    <- libraryP   d
+    isExecutable <- execP      d
+    isTest       <- testP      d
+    isBenchmark  <- benchmarkP d
+    pure Targets{..}
+
+githubP :: Decision -> Parser Decision
+githubP d = flag Idk d
+          $ long "github"
+         <> short 'g'
+         <> help "GitHub integration"
+
+travisP :: Decision -> Parser Decision
+travisP d = flag Idk d
+          $ long "travis"
+         <> short 'c'
+         <> help "Travis CI integration"
+
+appVeyorP :: Decision -> Parser Decision
+appVeyorP d = flag Idk d
+            $ long "app-veyor"
+           <> short 'w'
+           <> help "AppVeyor CI integration"
+
+privateP :: Decision -> Parser Decision
+privateP d = flag Idk d
+           $ long "private"
+          <> short 'p'
+          <> help "Private repository"
+
+scriptP :: Decision -> Parser Decision
+scriptP d = flag Idk d
+          $ long "script"
+         <> short 's'
+         <> help "Build script for convenience"
+
+libraryP :: Decision -> Parser Decision
+libraryP d = flag Idk d
+           $ long "library"
+          <> short 'l'
+          <> help "Library folder"
+
+execP :: Decision -> Parser Decision
+execP d = flag Idk d
+        $ long "exec"
+       <> short 'e'
+       <> help "Executable target"
+
+testP :: Decision -> Parser Decision
+testP d =  flag Idk d
+        $  long "test"
+        <> short 't'
+        <> help "Test target"
+
+benchmarkP :: Decision -> Parser Decision
+benchmarkP d = flag Idk d
+             $ long "benchmark"
+            <> short 'b'
+            <> help "Benchmarks"
+
+onP :: Parser Targets
+onP  = subparser $ mconcat
+       [ metavar "on [OPTIONS]"
+       , command "on" $ info (helper <*> targetsP Yes) (progDesc "Specify options to enable")
+       ]
+
+offP :: Parser Targets
+offP = subparser $ mconcat
+       [ metavar "off [OPTIONS]"
+       , command "off" $ info (helper <*> targetsP Nop) (progDesc "Specify options to disable")
+       ]
+
+optsP :: Parser InitOpts
+optsP = do
+    projectName <- T.pack <$> strArgument (metavar "PROJECT_NAME")
```

Comment:
I think you don't need `T.pack` here anymore. Latest version of `optparse-applicative` has polymorphic `strArgument` if I recall correctly.